books/views.py

```diff
@@ -48,3 +53,44 @@ def read_or_update(request, book_id):
     else:
         form = BookForm(instance=book)
         return render(request, 'update.html', {'book': book, 'form': form})
+
+
+class AuthorView(APIView):
+    def get(self, request):
+        authors = Author.objects.all()
+        serializer = AuthorSerializer(authors, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        author = request.data.get('author')
+        serializer = AuthorSerializer(data=author)
+        if serializer.is_valid():
+            try:
+                serializer.save()
+            except IntegrityError:
+                return Response({"message": "Integrity Error"}, status=status.HTTP_400_BAD_REQUEST)
+            return Response({
+                "message": "successfully created author",
+                "data": serializer.data
+            }, status=status.HTTP_201_CREATED)
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class AuthorDetail(APIView):
+    def get(self, request, pk, format=None):
+        author = get_object_or_404(Author.objects.all(), pk=pk)
```

Comment:
we shouldn't query `all()` when there is only one requested.

Reply:
acknowledged for all cases

Reply:
did not get?

Reply:
i will fix everywhere in code :)

Reply:
done